core/indigo-core/tests/common.cpp

```diff
@@ -75,3 +77,40 @@ std::string IndigoCoreTest::dataPath(const char* dataPathSuffix)
 {
     return dataPathPrefix + "/" + dataPathSuffix;
 }
+
+void IndigoCoreTest::loadReaction(const char* buf, Reaction& reaction)
+{
+    BufferScanner scanner(buf);
+    ReactionAutoLoader loader(scanner);
+    loader.loadReaction(reaction);
+}
+
+void IndigoCoreTest::loadQueryReaction(const char* buf, QueryReaction& queryReaction)
+{
+    BufferScanner scanner(buf);
+    ReactionAutoLoader loader(scanner);
+    loader.loadQueryReaction(queryReaction);
+}
+
+std::string IndigoCoreTest::saveReaction(Reaction& reaction)
+{
+    Array<char> buffer;
+    ArrayOutput output(buffer);
+    RSmilesSaver smilesSaver(output);
+    smilesSaver.saveReaction(reaction);
+    output.writeChar(0);
+    std::string result(buffer.ptr());
+    return result;
+}
+
+std::string IndigoCoreTest::saveQueryReaction(QueryReaction& queryReaction, const bool smarts)
```

Comment:
can we have the same saveReactionSmiles (function overloading)?

Reply:
Done